### PR TITLE
Ka ai tutor available component logic

### PR DESCRIPTION
**Pull request by kakiha11**
Updated the logic for the AI Tutor component based on whether a level has been marked as an assessment and if AI Tutor has been made available on that level by the level editor.

The AI Tutor should be available on a specific level if the level is NOT an assessment and AI Tutor HAS been marked as available on that level by a level builder. There are three examples of levels with different settings below to show it's working as expected.

A level that is NOT an assessment and does NOT have AI Tutor made available on that level by the level editor.

image

A level that is an assessment and the level editor has made AI Tutor available on that level.

Screenshot 2024-03-05 at 9 15 20 PM

A level that is NOT an assessment and AI Tutor has been marked as available on that level.

image

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/levels/show.html.haml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 - if @script_level
   - level_data = { redirect_script_url: @redirect_unit_url, script_name: @script_level.script&.name, course_name: @script_level.script&.unit_group&.name, level_name: @level.name, tts_autoplay_enabled: @tts_autoplay_enabled, code_review_enabled: @code_review_enabled_for_level, show_unversioned_redirect_warning: @show_unversioned_redirect_warning}
-  - ai_tutor_data = {levelId: @level&.id, scriptId: @script_level.script&.id, type: @level&.type, hasValidation: !!@level&.properties["encrypted_validation"], isAssessment: @script_level&.assessment, isProjectBacked: !!@level&.properties["project_template_level_name"]}
+  - ai_tutor_data = {levelId: @level&.id, scriptId: @script_level.script&.id, type: @level&.type, hasValidation: !!@level&.properties["encrypted_validation"], isProjectBacked: !!@level&.properties["project_template_level_name"], levelbuilderTutorChecked: @level&.levelbuilder_ai_tutor_available?, isAssessment: @script_level&.assessment}
```
Comment (Erin007) [inline dashboard/app/views/levels/show.html.haml]: `levelbuilderTutorChecked` -> `aiTutorAvailable`

File: dashboard/app/models/levels/level.rb [hunk 2]
```diff
@@ -514,6 +514,10 @@ def display_as_unplugged?
     unplugged? || properties["display_as_unplugged"] == "true"
   end
 
+  def levelbuilder_ai_tutor_available?
+    properties["ai_tutor_available"] == "true"
+  end
```
Comment (Erin007) [inline dashboard/app/models/levels/level.rb:519]: `levelbuilder_ai_tutor_available?` -> `ai_tutor_available?`

File: apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx [hunk 3]
```diff
@@ -14,6 +14,11 @@ interface AITutorPanelProps {
 const AITutorPanel: React.FunctionComponent<AITutorPanelProps> = ({open}) => {
   const level = useAppSelector(state => state.aiTutor.level);
   const isAssessmentLevel = level?.isAssessment;
+  const isLevelbuilderChecked = level?.levelbuilderTutorChecked;
```
Comment (Erin007) [inline apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx]: `const isAvailable = level?.aiTutorAvailable`

File: apps/src/sites/studio/pages/levels/show.js [hunk 4]
```diff
@@ -61,14 +61,21 @@ function initPage() {
...
+      isAssessment,
+    } = aiTutorData;
     const level = {
       id: levelId,
       type: type,
       hasValidation: hasValidation,
-      isAssessment: isAssessment,
       isProjectBacked: isProjectBacked,
+      aiTutorAvailable,
+      isAssessment: isAssessment,
```
Comment (Erin007) [inline apps/src/sites/studio/pages/levels/show.js]: I think you're going to have a merge conflict with yourself with https://github.com/code-dot-org/code-dot-org/pull/57051. Let me know if you want extra eyes resolving. Same comment here: let's be consistent with the fields.

**Review by Erin007 — APPROVED** [review]
One tiny comment, but looks great! The logic and names are much clearer from earlier drafts 🎉 Thank you!

File: apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx [hunk 5]
```diff
@@ -14,6 +14,11 @@ interface AITutorPanelProps {
 const AITutorPanel: React.FunctionComponent<AITutorPanelProps> = ({open}) => {
   const level = useAppSelector(state => state.aiTutor.level);
   const isAssessmentLevel = level?.isAssessment;
+  const isAvailable = level?.aiTutorAvailable;
+  let aiTutorAvailable = true;
```
Comment (ebeastlake) [inline apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx]: I think we can just use
```
const aiTutorAvailable = !isAssessmentLevel && isAvailable;
```
Does that seem right?

File: apps/src/sites/studio/pages/levels/show.js [hunk 6]
```diff
@@ -61,14 +61,21 @@ function initPage() {
...
   if (hasScriptData('script[data-aitutordata]')) {
     const aiTutorData = getScriptData('aitutordata');
-    const {levelId, type, hasValidation, isAssessment, isProjectBacked} =
-      aiTutorData;
+    const {
+      levelId,
+      type,
+      hasValidation,
+      isProjectBacked,
+      aiTutorAvailable,
```
Comment (ebeastlake) [inline apps/src/sites/studio/pages/levels/show.js:69]: `aiTutorAvailable` is the flag coming from the backend? Sorry, I'm getting used between `isAvailable` and `aiTutorAvailable` 😬

File: apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx (hunk 5 above)
Comment (ebeastlake) [reply apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx]: I wonder if we can avoid confusion with variable names by using the ones below...
```
const isAssessmentLevel = level?.isAssessment;
const aiTutorAvailable = level?.aiTutorAvailable;
const renderAITutor = !isAssessmentLevel && isAvailable;
```

**Comment by ebeastlake** [comment]
Some nits on variable naming, but the general approach looks good!

File: apps/src/sites/studio/pages/levels/show.js (hunk 6 above)
Comment (kakiha11) [reply apps/src/sites/studio/pages/levels/show.js:69]: Yes, `aiTutorAvailable` is the name of the variable that gets defined in the `show.html.haml` using the helper method I added to the `level.rb` model

File: apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx (hunk 5 above)
Comment (kakiha11) [reply apps/src/code-studio/components/aiTutor/aiTutorPanel.tsx]: Did it!

**Comment by kakiha11** [comment]
> Some nits on variable naming, but the general approach looks good!

Going to merge this! Can revert if necessary.

**Comment by ebeastlake** [comment]
Looks good, thanks, @kakiha11!

**Pull request merged**